core/src/io/anuke/mindustry/world/blocks/power/PowerGraph.java

```diff
@@ -168,20 +169,24 @@ public void distributePower(float needed, float produced){
                         }
                     }
                 }
+                
+                amountOfPowerUsed += consumer.entity.power.satisfaction;
             }
         }
+        
+        return amountOfPowerUsed;
     }
 
-    public void update(){
+    public float update(){
```

Comment:
Why does this return a value? It doesn't seem appropriate that updating the graph returns an arbitrary float. The power consumed should be a field that is updated each `update()` instead and queried with a getter.